core/temporal/src/components/tz.rs

```diff
@@ -16,29 +16,14 @@ use super::calendar::CalendarProtocol;
 pub const TIME_ZONE_PROPERTIES: [&str; 3] =
     ["getOffsetNanosecondsFor", "getPossibleInstantsFor", "id"];
 
-/// A clonable `TzProtocol`
-pub trait TzProtocolClone {
-    /// Clones the current `TimeZoneProtocol`.
-    fn clone_box(&self) -> Box<dyn TzProtocol>;
-}
-
-impl<P> TzProtocolClone for P
-where
-    P: 'static + TzProtocol + Clone,
-{
-    fn clone_box(&self) -> Box<dyn TzProtocol> {
-        Box::new(self.clone())
-    }
-}
-
 /// The Time Zone Protocol that must be implemented for time zones.
-pub trait TzProtocol: TzProtocolClone {
+pub trait TzProtocol: Clone {
```

Comment:
Thought: We should probably add an associated error type `type Error: From<TemporalError>` so that we can preserve the full engine error. Would also have to be added to `CalendarProtocol`.

Reply:
That makes sense. As the change can be done on both traits. I'll update that on another PR for both traits.